OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -794,9 +794,8 @@ void GiveOrdersToIdleHarvesters()
 
 				if (!harvester.IsIdle)
 				{
-					var act = harvester.GetCurrentActivity();
-
-					if (act.NextActivity == null || act.NextActivity.GetType() != typeof(FindResources))
+					var act = harvester.CurrentActivity;
+					if (act.NextActivity != null && act.NextActivity.GetType() != typeof(FindResources))
```

Comment:
Changing the `||` to a `&&` seems to regress behaviour ingame: the AI spams new harvest orders every time this runs.